components/omega/src/timeStepping/TimeStepper.h

```diff
@@ -0,0 +1,126 @@
+#ifndef OMEGA_TIMESTEPPER_H
+#define OMEGA_TIMESTEPPER_H
+//===-- timeStepping/TimeStepper.h - time stepper --------------------*- C++
+//-*-===//
+//
+/// \file
+/// \brief Contains the base class for all Omega time steppers
+///
+/// The TimeStepper class defines the interface of a time stepper and contains
+/// data and methods common to all time steppers
+//
+//===----------------------------------------------------------------------===//
+
+#include "AuxiliaryState.h"
+#include "DataTypes.h"
+#include "Halo.h"
+#include "HorzMesh.h"
+#include "OceanState.h"
+#include "TendencyTerms.h"
+
+#include <map>
+#include <memory>
+#include <string>
+
+namespace OMEGA {
+
+/// An enum for every time stepper type
+/// needs to extended every time a new time stepper is added
+enum class TimeStepperType { ForwardBackward, RungeKutta4, RungeKutta2 };
+
+//------------------------------------------------------------------------------
+/// A base class for Omega time steppers
+///
+/// The TimeStepper class defines the interface of a time stepper, manages
+/// time stepper objects and contains common routines for state updates
+class TimeStepper {
+ public:
+   virtual ~TimeStepper() = default;
+
+   // The main method that every time stepper needs to define. Advances state by
+   // by one time step, from Time to Time + TimeStep
+   virtual void doStep(OceanState *State, Real Time, Real TimeStep) const = 0;
```

Comment:
I'm a little confused by the Time being included as an argument of the doStep function here, and in the various compute tendencies functions in the Tendency class. The tendency calculations are not generally dependent on the time directly, and the only thing I'm seeing it used for in the PR is for computing the exact solution in the unit test.

Reply:
Also, since the TimeStep is kind of core to the purpose of the TimeStepper, maybe makes sense to just store it as a member variable of the class

Reply:
The manufactured solution has forcings that depend explicitly on time. I agree about the `TimeStep`. Initially I thought that making it an argument will make it easier to support varying it during a simulation, but now I think that, if we want to support that, having a `changeTimeStep` method would be better.

Reply:
For down the road, tidal potential forcing also depends on time.

Reply:
I made the `TimeStep` a member, and changed `Time` and `TimeStep` to use types `TimeInstant` and `TimeInterval`, respectively.